packages/cli/src/handlers/dbt/getWarehouseClient.ts

```diff
@@ -0,0 +1,263 @@
+                project: '',
+                dataset: '',
+                timeoutSeconds: undefined,
+                priority: undefined,
+                keyfileContents: {},
+                retries: undefined,
+                location: undefined,
+                maximumBytesBilled: undefined,
+            };
+            break;
+        case SupportedDbtAdapter.POSTGRES:
+            credentials = {
+                type: WarehouseTypes.POSTGRES,
+                host: '',
+                user: '',
+                password: '',
+                port: 5432,
+                dbname: '',
+                schema: '',
+            };
+            break;
+        case SupportedDbtAdapter.REDSHIFT:
+            credentials = {
+                type: WarehouseTypes.REDSHIFT,
+                host: '',
+                user: '',
+                password: '',
+                port: 5432,
+                dbname: '',
+                schema: '',
+            };
+            break;
+        case SupportedDbtAdapter.SNOWFLAKE:
+            credentials = {
+                type: WarehouseTypes.SNOWFLAKE,
+                account: '',
+                user: '',
+                password: '',
+                warehouse: '',
+                database: '',
+                schema: '',
+                role: '',
+            };
+            break;
+
+        case SupportedDbtAdapter.DATABRICKS:
+            credentials = {
+                type: WarehouseTypes.DATABRICKS,
+                catalog: '',
+                database: '',
+                serverHostName: '',
+                httpPath: '',
+                personalAccessToken: '',
+            };
+            break;
+        case SupportedDbtAdapter.TRINO:
+            credentials = {
+                type: WarehouseTypes.TRINO,
+                host: '',
+                user: '',
+                password: '',
+                port: 5432,
+                dbname: '',
+                schema: '',
+                http_scheme: '',
+            };
+            break;
+        default:
+            assertUnreachable(
+                dbtAdaptorType,
+                `Unsupported dbt adaptor type ${dbtAdaptorType}`,
+            );
+    }
+    return credentials;
+}
+
+type GetWarehouseClientOptions = {
+    isDbtCloudCLI: boolean;
+    profilesDir: string;
+    profile: string;
+    target?: string;
+    startOfWeek?: number;
+};
+
+type GetWarehouseClientReturn = {
+    warehouseClient: ReturnType<typeof warehouseClientFromCredentials>;
+    credentials: CreateWarehouseCredentials;
+};
+
+export default async function getWarehouseClient(
+    options: GetWarehouseClientOptions,
+): Promise<GetWarehouseClientReturn> {
+    let warehouseClient;
+    let credentials;
+    if (options.isDbtCloudCLI) {
+        const dbtAdaptorType = await getDbtCloudConnectionType();
+        GlobalState.debug(`> Using ${dbtAdaptorType} client mock`);
+        credentials = getMockCredentials(dbtAdaptorType);
```

Comment:
For dbt cloud, we generate empty credentials for the correct warehouse. And then we override the warehouse client methods that need conection to the warehouse.
Only the `getCatalog` needs a workaround implementation so we can generate the yml files based on the schema.